File: app/src/main/java/com/example/ballrsrv/CourtsAdapter.java
package com.example.ballrsrv;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import java.util.List;
import java.util.Locale;

public class CourtsAdapter extends RecyclerView.Adapter<CourtsAdapter.CourtViewHolder> {
    private List<Court> courtsList;
    private OnCourtActionListener listener;

    public interface OnCourtActionListener {
        void onRemoveCourt(Court court);
    }

    public CourtsAdapter(List<Court> courtsList, OnCourtActionListener listener) {
        this.courtsList = courtsList;
        this.listener = listener;
    }

    @NonNull
    @Override
    public CourtViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_court, parent, false);
        return new CourtViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CourtViewHolder holder, int position) {
        Court court = courtsList.get(position);
    }

    @Override
    public int getItemCount() {
        return courtsList.size();
    }


            super(itemView);
            courtImage = itemView.findViewById(R.id.courtImage);
            btnRemove = itemView.findViewById(R.id.btnRemove);
        }
    }
} 

File: app/src/main/java/com/example/ballrsrv/adapters/CourtsAdapter.kt
package com.example.ballrsrv.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.ballrsrv.data.Court
import com.example.ballrsrv.databinding.ItemCourtBinding
import com.squareup.picasso.Picasso

class CourtsAdapter(
    private val courts: List<Court>,
    private val onCourtClick: (Court) -> Unit
) : RecyclerView.Adapter<CourtsAdapter.CourtViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CourtViewHolder {
        val binding = ItemCourtBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return CourtViewHolder(binding)
    }

    override fun onBindViewHolder(holder: CourtViewHolder, position: Int) {
        holder.bind(courts[position])
    }

    override fun getItemCount(): Int = courts.size

    inner class CourtViewHolder(
        private val binding: ItemCourtBinding
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(court: Court) {
            binding.apply {
                courtName.text = court.name
                courtDescription.text = court.description
                courtPrice.text = "₱${court.price}"
                
                court.imageUrl?.let { url ->
                    Picasso.get().load(url).into(courtImage)
                }

                root.setOnClickListener { onCourtClick(court) }
            }
        }
    }
} 